File: Element Appearing More Than 25% In Sorted Array/solution.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findSpecialInteger = function (arr) {
    let curr = 1;
    let lastIndex = -1;
    let compare = arr[0];
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] === compare) curr++;
        else {
            lastIndex = i;
            break;
        }
    }
    if (curr > Math.floor(arr.length / 4)) return compare;
    compare = arr[lastIndex];
    curr = 1;
    for (let i = lastIndex + 1; i < arr.length; i++) {
        if (arr[i] === compare) curr++;
        else {
            if (curr > Math.floor(arr.length / 4)) return compare;
            compare = arr[i];
            curr = 1;
        }
    }
    if (curr > Math.floor(arr.length / 4)) return compare;
};

File: Element Appearing More Than 25% In Sorted Array/solution.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        curr = 1
        compare = arr[0]
        last_index = -1
        for i in range(1, len(arr)):
            if arr[i] == compare:
                curr += 1
            else:
                last_index = i
                break
        if curr > len(arr) // 4:
            return compare
        compare = arr[last_index]
        curr = 1
        for i in range(last_index+1, len(arr)):
            if arr[i] == compare:
                curr += 1
            else:
                if curr > len(arr) // 4:
                    return compare
                compare = arr[i]
                curr = 1
        if curr > len(arr) // 4:
            return compare